client/my-sites/stats/stats-date-control/stats-date-control-picker.tsx

```diff
@@ -7,18 +7,67 @@ import DateControlPickerShortcuts from './stats-date-control-picker-shortcuts';
 import { DateControlPickerProps } from './types';
 
 const DateControlPicker = ( { slug, queryParams }: DateControlPickerProps ) => {
+	const redirectToQueryDate = ( daysToSubtract: number ) => {
+		const startDate = calculateQueryDate( daysToSubtract );
+		const query = qs.stringify( Object.assign( {}, queryParams, { startDate } ), {
+			addQueryPrefix: true,
+		} );
+		const period = 'day'; // TODO: adjust this as needed
+		const url = `/stats/${ period }/${ slug }`;
+		const href = `${ url }${ query }`;
+
+		page( href );
+	};
```

Comment:
Instead of updating the page URL, I think we just want to update the text input field. So something like this maybe:

```
const refreshStartDate = ( daysToSubtract: number ) => {
	const newStartDate = calculateQueryDate( daysToSubtract );
	changeStartDate( newStartDate );
}
```

In that way we prevent changes from being applied unless the "Apply" button is clicked.